pageserver/src/http/openapi_spec.yml

```diff
@@ -928,12 +928,29 @@ components:
               writing to the tenant's S3 state, so, DO NOT ATTACH the
               tenant to any other pageserver, or we risk split-brain.
             - `attached` means that the attach operation has completed,
-              maybe successfully, maybe not. Perform a health check at
-              the Postgres level to determine healthiness of the tenant.
+              successfully
+            - `failed` means that attach has failed. For reason check corresponding `reason` failed.
+              `failed` is the terminal state, retrying attach call wont resolve the issue.
+              For example this can be caused by s3 being unreachable. The retry may be implemented
+              with call to detach, though it would be better to not automate it and inspect failed state
+              manually before proceeding with a retry.
 
             See the tenant `/attach` endpoint for more information.
-          type: string
-          enum: [ "maybe", "attached" ]
+          oneOf:
+            - type: string
+              enum: [ "maybe", "attached" ]
+            - type: object
+              required:
+                - failed
+              properties:
+                failed:
+                  type: object
+                  required:
+                    - reason
+                  properties:
+                    reason:
+                      type: string
```

Comment:
This is so ugly, why not breaking-change the `AttachmentStatus` serde to the `slug`-style which we use for TenantStatus?

It's ok to break the relocation API, it's not in use yet.

Reply:
Ah, the problem is that breaking relocation API will break the e2e tests in this PR.
Still, I think we should use slug-style here.
How do work around the e2e tests?

Reply:
Decided to convert it to slug/data api as it is for tenant status, to work around e2e tests we will temporarily disable relocation test in cloud repo.